torch/testing/_internal/common_jit.py

```diff
@@ -48,13 +49,28 @@ def allSum(vs):
                    if v is not None and v.dtype in floating_and_complex_types_and(torch.half, torch.bfloat16))
 
     def clone_inputs(requires_grad):
-        inputs = [
-            arg.detach().clone().requires_grad_(requires_grad and arg.requires_grad)
-            if isinstance(arg, torch.Tensor) else arg for arg in args
-        ]
-        return inputs, [input for input in inputs if isinstance(input, torch.Tensor) and input.requires_grad]
+        inputs: List[Union[torch.Tensor, List[torch.Tensor]]] = []
+        recording_tensors: List[torch.Tensor] = []
+        for arg in args:
+            if isinstance(arg, torch.Tensor):
+                require_grad = requires_grad and arg.requires_grad
```

Comment:
This name being so similar to the input `requires_grad` name makes reading this a little rough.

Reply:
`should_record`?